apps/experiments/views/experiment.py

```diff
@@ -567,74 +568,14 @@ def update_delete_channel(request, team_slug: str, experiment_id: int, channel_i
-    participant_user: CustomUser | None = None,
-    session_status: SessionStatus = SessionStatus.ACTIVE,
-    timezone: str | None = None,
-) -> ExperimentSession:
-    if not participant_identifier and not participant_user:
-        raise ValueError("Either participant_identifier or participant_user must be specified!")
-
-    if participant_user and participant_identifier != participant_user.email:
-        # This should technically never happen, since we disable the input for logged in users
-        raise Exception(f"User {participant_user.email} cannot impersonate participant {participant_identifier}")
-
-    with transaction.atomic():
-        try:
-            participant = Participant.objects.get(team=experiment.team, identifier=participant_identifier)
-            if participant_user and participant.user is None:
-                # If a participant becomes a user, we must reconcile the user and participant
-                participant.user = participant_user
-                participant.save()
-        except Participant.DoesNotExist:
-            participant = Participant.objects.create(
-                user=participant_user,
-                identifier=participant_identifier,
-                team=experiment.team,
-            )
-        session = ExperimentSession.objects.create(
-            team=experiment.team,
-            experiment=experiment,
-            llm=experiment.llm,
-            experiment_channel=experiment_channel,
-            status=session_status,
-            participant=participant,
-        )
-
-        # Record the participant's timezone
-        if timezone:
-            data_records = ParticipantData.objects.filter(participant=participant)
-            for data_record in data_records:
-                data_record.data["timezone"] = timezone
-                data_record.save()
-            ParticipantData.objects.bulk_update(data_records, fields=["data"])
-
-    if participant.experimentsession_set.count() == 1:
-        enqueue_static_triggers.delay(session.id, StaticTriggerType.PARTICIPANT_JOINED_EXPERIMENT)
-    enqueue_static_triggers.delay(session.id, StaticTriggerType.CONVERSATION_START)
-    return _check_and_process_seed_message(session)
-
-
-def _check_and_process_seed_message(session: ExperimentSession):
-    if session.experiment.seed_message:
-        session.seed_task_id = get_response_for_webchat_task.delay(
-            session.id, message_text=session.experiment.seed_message
-        ).task_id
-        session.save()
-    return session
-
-
 @require_POST
 @login_and_team_required
 def start_authed_web_session(request, team_slug: str, experiment_id: int):
     experiment = get_object_or_404(Experiment, id=experiment_id, team=request.team)
     experiment_channel = _ensure_experiment_channel_exists(
         experiment=experiment, platform="web", name=f"{experiment.id}-web"
     )
-    session = _start_experiment_session(
+    session = WebChannel.start_new_session(
```

Comment:
I like this change. We can move the `experiment_channel = _ensure_experiment_channel_exists(...)` into the `start_new_session` method as well, since we know that the instance should always be a webchannel instance. Might as well create it in there. But maybe for a different PR. I'll take note of this